test/screenshot/menu-surface/index.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+
+import '../../../packages/menu-surface/index.scss';
+import '@material/list/mdc-list.scss';
+import './index.scss';
+
+import MenuSurface, {Corner} from '../../../packages/menu-surface/index';
+import Button from '../../../packages/button/index';
+
+class MenuSurfaceButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: props.open, // eslint-disable-line react/prop-types
+      coordinates: null,
+    };
+  }
+
+  componentDidMount() {
+    this.rightClickCallback_ = (evt) => {
+      this.setState({
+        open: true,
+        coordinates: {x: evt.clientX, y: evt.clientY},
+      });
+      evt.preventDefault();
+    };
+
+    if (this.props.contextmenu) { // eslint-disable-line react/prop-types
+      window.addEventListener('contextmenu', this.rightClickCallback_);
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('contextmenu', this.rightClickCallback_);
+  }
+
+  setAnchorElement = (element) => {
```

Comment:
I want the component to re-render when the ref is set. I put it on state, so that it renders again once it sets. Otherwise I did a forceRender(), which isn't as obvious.